lib/base/tlsutility.cpp

```diff
@@ -640,12 +640,16 @@ int MakeX509CSR(const String& cn, const String& keyfile, const String& csrfile,
 	return 1;
 }
 
-std::shared_ptr<X509> CreateCert(EVP_PKEY *pubkey, X509_NAME *subject, X509_NAME *issuer, EVP_PKEY *cakey, bool ca)
+std::shared_ptr<X509> CreateCert(EVP_PKEY *pubkey, X509_NAME *subject, X509_NAME *issuer, EVP_PKEY *cakey, long validFor, bool ca)
```

Comment:
`long` looks like a bit of an odd choice here, given that part of this PR relates to problems because `time_t` might be 32 bit only and `long` has the very same property that it's typically only 32 bits on 32 bit systems. (It still works as it's only used to pass values relative to the current time, as long as they don't exceed that value range, we don't seem to care for more than 15 years here.)

Reply:
Ok, then there's no point in using any larger data type if we couldn't even pass it to OpenSSL. Interesting choice of type on their end though.